clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/metrics/UrlMetricsInstrumentation.groovy

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.docker.registry.metrics
+
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+import com.squareup.okhttp.Response
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+import java.time.Duration
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+import static java.lang.String.format
+
+@Component
+class UrlMetricsInstrumentation {
+
+  private Logger log = LoggerFactory.getLogger(UrlMetricsInstrumentation)
+
+  private final Registry registry
+
+  private final Id timingId
+  private final Id counterId
+
+  @Autowired
+  UrlMetricsInstrumentation(Registry registry) {
+    this.registry = registry
+    timingId = registry.createId('UrlRequestTime').withTag('className', UrlMetricsInstrumentation.simpleName)
+    counterId = registry.createId('UrlCount').withTag('className', UrlMetricsInstrumentation.simpleName)
+  }
```

Comment:
The spinnaker timer metrics have both a count and a time, so this UrlCount is superfluous.

In the timer, there is a "statistic" label which is either "count" or "totalTime". The "count" acts as a call count, the "totalTime" is the time counter. I dont like that it works like that but it is what it is. If you are using the monitoring daemon, it will split a timers into the two separate counters *__count and *__totalTime.